lib/ezpool/connection_wrapper.rb

```diff
@@ -7,14 +7,19 @@ def initialize(conn, connection_manager)
     @raw_conn = conn
     @created_at = EzPool.monotonic_time
     @manager = connection_manager
+    @expired = false
   end
 
   # Shut down the connection. Can no longer be used after this!
   def shutdown!
     @manager.disconnect(@raw_conn)
   end
 
+  def expire!
+    @expired = true
+  end
+
   def age
-    EzPool.monotonic_time - @created_at
+    EzPool.monotonic_time - @created_at || @expired == true
```

Comment:
this doesn't make much sense to me; `age` returns an integer which is compared against `max_age`.

Maybe you want to change the `expired?` method in `EzPool` to check the expired property of the connection wrapper?

Reply:
This was a goof and i ended up doing almost exactly what you suggested.